flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-doris/src/main/java/com/ververica/cdc/connectors/doris/factory/DorisDataSinkFactory.java

```diff
@@ -0,0 +1,148 @@
+        DorisExecutionOptions.Builder executionBuilder = DorisExecutionOptions.builderDefaults();
+        config.getOptional(FENODES).ifPresent(optionsBuilder::setFenodes);
+        config.getOptional(BENODES).ifPresent(optionsBuilder::setBenodes);
+        config.getOptional(TABLE_IDENTIFIER).ifPresent(optionsBuilder::setTableIdentifier);
+        config.getOptional(USERNAME).ifPresent(optionsBuilder::setUsername);
+        config.getOptional(PASSWORD).ifPresent(optionsBuilder::setPassword);
+        config.getOptional(JDBC_URL).ifPresent(optionsBuilder::setJdbcUrl);
+        config.getOptional(AUTO_REDIRECT).ifPresent(optionsBuilder::setAutoRedirect);
+
+        config.getOptional(SINK_CHECK_INTERVAL).ifPresent(executionBuilder::setCheckInterval);
+        config.getOptional(SINK_MAX_RETRIES).ifPresent(executionBuilder::setMaxRetries);
+        config.getOptional(SINK_ENABLE_DELETE).ifPresent(executionBuilder::setDeletable);
+        config.getOptional(SINK_LABEL_PREFIX).ifPresent(executionBuilder::setLabelPrefix);
+        config.getOptional(SINK_BUFFER_SIZE).ifPresent(executionBuilder::setBufferSize);
+        config.getOptional(SINK_BUFFER_COUNT).ifPresent(executionBuilder::setBufferCount);
+        config.getOptional(SINK_BUFFER_FLUSH_MAX_ROWS)
+                .ifPresent(executionBuilder::setBufferFlushMaxRows);
+        config.getOptional(SINK_BUFFER_FLUSH_MAX_BYTES)
+                .ifPresent(executionBuilder::setBufferFlushMaxBytes);
+        config.getOptional(SINK_FLUSH_QUEUE_SIZE).ifPresent(executionBuilder::setFlushQueueSize);
+        config.getOptional(SINK_IGNORE_UPDATE_BEFORE)
+                .ifPresent(executionBuilder::setIgnoreUpdateBefore);
+        config.getOptional(SINK_USE_CACHE).ifPresent(executionBuilder::setUseCache);
+        config.getOptional(SINK_BUFFER_FLUSH_INTERVAL)
+                .ifPresent(v -> executionBuilder.setBufferFlushIntervalMs(v.toMillis()));
+
+        config.getOptional(SINK_ENABLE_2PC)
+                .ifPresent(
+                        b -> {
+                            if (b) {
+                                executionBuilder.enable2PC();
+                            } else {
+                                executionBuilder.disable2PC();
+                            }
+                        });
+        config.getOptional(SINK_ENABLE_BATCH_MODE)
+                .ifPresent(
+                        b -> {
+                            if (b) {
+                                executionBuilder.enableBatchMode();
+                            }
+                        });
+        return new DorisDataSink(
+                optionsBuilder.build(),
+                DorisReadOptions.builder().build(),
+                executionBuilder.build());
+    }
+
+    @Override
+    public String identifier() {
+        return "doris";
+    }
+
+    @Override
+    public Set<ConfigOption<?>> requiredOptions() {
+        return new HashSet<>();
```

Comment:
In my test case,  met the follow exception when FENODES was not defined. So is it required?

```
Exception in thread "main" java.lang.NullPointerException: No fenodes supplied.
	at org.apache.flink.util.Preconditions.checkNotNull(Preconditions.java:76)
	at org.apache.doris.flink.cfg.DorisOptions$Builder.build(DorisOptions.java:134)
	at com.ververica.cdc.connectors.doris.factory.DorisDataSinkFactory.createDataSink(DorisDataSinkFactory.java:104)
```

Reply:
Thanks for the reminder, it has been modified